mods/gen2/moves.js

```diff
@@ -513,9 +513,9 @@ exports.BattleMovedex = {
 	roar: {
 		inherit: true,
 		onTryHit: function () {
-			for (let i = 0; i < this.queue.length; i++) {
+			for (const action of this.queue) {
 				// Roar only works if it is the last action in a turn, including when it's called by Sleep Talk
-				if (this.queue[i].choice === 'move' || this.queue[i].choice === 'switch') return false;
+				if (['move', 'switch'].includes(action.choice)) return false;
```

Comment:
Heyy, `includes` is a refactor for the old `in {object}` code. `||` is still more performant, and should be done for short tests like this one.